pywbemtools/pywbemcli/_display_cimobjects.py

```diff
@@ -478,10 +478,45 @@ def _display_instances_as_table(insts, table_width, table_format,
     else:
         max_cell_width = table_width
 
-    header_line = []
+    # TODO: Decide whether and how the showing of units should be controlled.
+    show_units = True
+
+    if show_units:
+        # Retrieve the most derived class of these instances. This is necessary
+        # because that is the only class that has all properties that may be
+        # needed.
+        derived_class_obj = None
+        for inst in insts:
+            classname = inst.classname
+            if inst.path is None:
+                # The only operation returning instances without a path is
+                # query execution. For now, we simply don't display units in
+                # that case.
+                # TODO: Pass a namespace for query execution to display units
+                continue
+            namespace = inst.path.namespace
+            class_obj = context.conn.GetClass(
```

Comment:
This could get the same class many times.   Should we cache the class to be reused?

Reply:
DONE - the new implementation that addresses the second thoughts stated above gets each class only once.